### PR TITLE
prepare data

### PR DESCRIPTION
**Pull request by lynxbyorion**

### CODE REVIEW

File: collector/config.yaml [hunk 1]
```diff
@@ -1,3 +1,13 @@
 server:
   port: 8080
   payload_max_size: 5120
+service:
+  max_len_chunk: 2
+  chunk_size: 1024
```
Comment (stuffy-idler) [inline collector/config.yaml:6]: Two other parameters called as `max_<smth>`, maybe this one should be called by analogy?

File: collector/src/main.rs [hunk 2]
```diff
@@ -34,29 +28,86 @@ async fn main() -> std::io::Result<()> {
     };
     info!("config loaded"; "config" => &app_config);
 
+    let (sender, reciver) = mpsc::channel::<store::Request>(32);
```
Comment (stuffy-idler) [inline collector/src/main.rs:31]: Mistype: missing `e` in the `receiver`

File: collector/src/main.rs [hunk 3]
```diff
@@ -34,29 +28,86 @@ async fn main() -> std::io::Result<()> {
...
+        sender,
+    });
+
+    tokio::spawn(async move { process(reciver, &app_config).await });
 
     HttpServer::new(move || {
         App::new()
             .wrap(Logger::default())
-            .app_data(app_state.clone())
+            .app_data(data.clone())
```
Comment (stuffy-idler) [inline collector/src/main.rs:45]: Is it necessary `data` to be cloned?

File: collector/src/main.rs [hunk 4]
```diff
@@ -34,29 +28,86 @@ async fn main() -> std::io::Result<()> {
...
 }
 
-async fn index(req: HttpRequest, body: Bytes, state: web::Data<Service>) -> impl Responder {
-    println!("url: {}", req.path());
-    println!("body: {:?}", body);
-
-    if body.len() > state.app_config.server.payload_max_size {
+async fn index(req: HttpRequest, body: web::Bytes, state: web::Data<AppState>) -> impl Responder {
+    if body.len() > state.config.server.payload_max_size {
         return "NOT OK";
```
Comment (stuffy-idler) [inline collector/src/main.rs:55]: Let's add logging for this case, I think `warn` should be suitable

File: collector/src/main.rs [hunk 5]
```diff
@@ -34,29 +28,86 @@ async fn main() -> std::io::Result<()> {
...
+                None => String::new(),
+            },
+            method: req.method().to_string(),
+            path: req.uri().to_string(),
+            headers: req
+                .headers()
+                .iter()
+                .map(|x| -> (String, String) {
+                    (
+                        x.0.to_string(),
```
Comment (stuffy-idler) [inline collector/src/main.rs:76]: I suggest to destructarize `x` at closure parameters with appropriate names

File: collector/src/main.rs [hunk 6]
```diff
@@ -34,29 +28,86 @@ async fn main() -> std::io::Result<()> {
...
     "OK"
 }
+
+async fn process(mut reciver: mpsc::Receiver<store::Request>, config: &AppConfig) {
+    let mut store = store::Store::new(config.service.chunk_size, config.service.max_len_chunk);
+
+    let mut delay = tokio::time::interval(config.service.max_collect_chunk_duration.into());
+
+    loop {
+        delay.reset();
```
Comment (stuffy-idler) [inline collector/src/main.rs:100]: Whether path `_ = delay.tick()` is reachable with resetting interval at each loop iteration if the receiver will receive messages more often then `max_collect_chunk_duration`?

File: collector/src/store.rs [hunk 7]
```diff
@@ -0,0 +1,79 @@
+use deepsize::DeepSizeOf;
```
Comment (stuffy-idler) [inline collector/src/store.rs:1]: Let's add tests for the methods in this module

**Review by stuffy-idler — CHANGES_REQUESTED** [review]

File: collector/src/store.rs (hunk 7 above)
Comment (lynxbyorion) [reply collector/src/store.rs:1]: Add tests in the next issue. Because Kafka sender is desable.

File: collector/src/main.rs (hunk 2 above)
Comment (lynxbyorion) [reply collector/src/main.rs:31]: fixed

File: collector/src/main.rs (hunk 3 above)
Comment (lynxbyorion) [reply collector/src/main.rs:45]: Seem to be. Closures are lazy, we do not know the lifetime of the variables in them.

File: collector/src/main.rs (hunk 4 above)
Comment (lynxbyorion) [reply collector/src/main.rs:55]: Fixed

File: collector/src/main.rs (hunk 5 above)
Comment (lynxbyorion) [reply collector/src/main.rs:76]: I agree. Fixed.

File: collector/src/main.rs (hunk 6 above)
Comment (lynxbyorion) [reply collector/src/main.rs:100]: hmm. It's wrong logic. Fixed.

File: collector/src/store.rs (hunk 7 above)
Comment (stuffy-idler) [reply collector/src/store.rs:1]: Got it. Ok

**Review by stuffy-idler — APPROVED** [review]

**Pull request merged**